### ISSUE TITLE
Bug: npm package was not found @typescript-eslint/shared-fixtures

### ISSUE THREAD

**Issue by mengfanliao** [typescript-eslint/typescript-eslint#5416]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Repro Code

```typescript
when I install @typescript-eslint/parser，got a error.
error Couldn't find package "@typescript-eslint/shared-fixtures" on the "npm" registry.
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

 @typescript-eslint/shared-fixtures can search and installed in npm

### Actual Result

npm package was not found @typescript-eslint/shared-fixtures

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.32.0` |
| `@typescript-eslint/parser`            | `5.32.0` |

**Comment by bradzacher** [comment]
The package is listed as a dev dependency, so there is no way npm is installing this as part of a standard install.

Could you please provide more information? Where are. You running the command? What command are you running? What are you doing? Help me to help you.
